editor/src/components/inspector/sections/component-section/variables-in-scope-utils.ts

```diff
@@ -322,102 +320,98 @@ function createPropertyAccessExpressionString(expressionSoFar: string, toAppend:
   }
 }
 
-function getTargetPropertyFromControlDescription(
-  controlDescription: ControlDescription | null,
-  targetPropertyName: string | null,
-): ControlDescription | null {
-  if (targetPropertyName == null || controlDescription == null) {
-    return controlDescription
-  } else {
-    switch (controlDescription.control) {
-      case 'object':
-        return controlDescription.object[targetPropertyName] ?? null
-      default:
-        return controlDescription
-    }
-  }
-}
+type Matcher = (_: VariableInfo) => VariableMatches | null
 
-export function orderVariablesForRelevance(
+function matchVariables(
   variableNamesInScope_MUTABLE: Array<VariableInfo>,
-  controlDescription: ControlDescription | null,
-  currentPropertyValue: PropertyValue,
-  targetPropertyName: string | null,
+  matchers: Array<Matcher>,
 ): Array<VariableInfo> {
-  let valuesExactlyMatchingPropertyName: Array<VariableInfo> = []
-  let valuesExactlyMatchingPropertyDescription: Array<VariableInfo> = []
-  let valuesMatchingPropertyDescription: Array<VariableInfo> = []
-  let valuesMatchingPropertyShape: Array<VariableInfo> = []
-  let valueElementMatches: Array<VariableInfo> = []
-  let restOfValues: Array<VariableInfo> = []
-
-  for (let variable of variableNamesInScope_MUTABLE) {
+  let categories: Array<VariableInfo[]> = matchers.map(() => [])
+  let rest: Array<VariableInfo> = []
+
+  const matchersWithIndex: Array<[Matcher, number]> = matchers.map((m, i) => [m, i])
+
+  variableNamesInScope_MUTABLE.forEach((variable) => {
     if (variable.type === 'array') {
-      variable.elements = orderVariablesForRelevance(
-        variable.elements,
-        controlDescription,
-        currentPropertyValue,
-        targetPropertyName,
-      )
+      variable.elements = matchVariables(variable.elements, matchers)
     } else if (variable.type === 'object') {
-      variable.props = orderVariablesForRelevance(
-        variable.props,
-        controlDescription,
-        currentPropertyValue,
-        targetPropertyName,
-      )
+      variable.props = matchVariables(variable.props, matchers)
+    }
+
+    for (const [matcher, idx] of matchersWithIndex) {
```

Comment:
ultra nitpick but I think a standard for loop is a more idiomatic way to iterate over an array while getting an index :)

Reply:
I replaced it with `matchers.entries()` (which seems to be the recommened way to do this)